lib/req_ch.ex

```diff
@@ -1,12 +1,167 @@
 defmodule ReqCh do
   @moduledoc """
   A Req plugin for ClickHouse.
+
+  By default, `ReqCh` will use TSV as the default output format.
+  To change that, see the `attach/2` docs for details.
   """
 
+  @connection_options [
+    :database,
+    :scheme,
+    :username,
+    :password,
+    :hostname,
+    :port
+  ]
+
+  @general_options [
+    :clickhouse,
+    :format
+  ]
+
+  @formats_page "https://clickhouse.com/docs/en/interfaces/formats"
+  @supported_formats ~w(TabSeparated TabSeparatedRaw TabSeparatedWithNames TabSeparatedWithNamesAndTypes TabSeparatedRawWithNames TabSeparatedRawWithNamesAndTypes Template TemplateIgnoreSpaces CSV CSVWithNames CSVWithNamesAndTypes CustomSeparated CustomSeparatedWithNames CustomSeparatedWithNamesAndTypes SQLInsert Values Vertical JSON JSONAsString JSONAsObject JSONStrings JSONColumns JSONColumnsWithMetadata JSONCompact JSONCompactStrings JSONCompactColumns JSONEachRow PrettyJSONEachRow JSONEachRowWithProgress JSONStringsEachRow JSONStringsEachRowWithProgress JSONCompactEachRow JSONCompactEachRowWithNames JSONCompactEachRowWithNamesAndTypes JSONCompactStringsEachRow JSONCompactStringsEachRowWithNames JSONCompactStringsEachRowWithNamesAndTypes JSONObjectEachRow BSONEachRow TSKV Pretty PrettyNoEscapes PrettyMonoBlock PrettyNoEscapesMonoBlock PrettyCompact PrettyCompactNoEscapes PrettyCompactMonoBlock PrettyCompactNoEscapesMonoBlock PrettySpace PrettySpaceNoEscapes PrettySpaceMonoBlock PrettySpaceNoEscapesMonoBlock Prometheus Protobuf ProtobufSingle ProtobufList Avro AvroConfluent Parquet ParquetMetadata Arrow ArrowStream ORC One Npy RowBinary RowBinaryWithNames RowBinaryWithNamesAndTypes RowBinaryWithDefaults Native Null XML CapnProto LineAsString Regexp RawBLOB MsgPack MySQLDump DWARF Markdown Form)
+
+  import Req.Request, only: [get_option: 3]
+
   @doc """
   Attach this plugin to a Req Request.
+
+  ## Options
+
+    * `:scheme` - Required. Either `"http"` or `"https"`. Default is `"http"`.
+    * `:username` - Optional. Default is `nil`.
+    * `:password` - Optional. Default is `nil`.
+    * `:hostname` - Required. Default is `"localhost"`.
+    * `:port` - Required. Default is `"8123"`.
+    * `:clickhouse` - Required. The query to be performed. If not provided,
+      the request and response pipelines won't be modified.
+    * `:format` - Optional. The format of the response. Default is `:tsv`.
+      This option accepts `:tsv`, `:csv` or `:explorer` as atoms.
+
+      It also accepts all formats described in the #{@formats_page} page.
+
+      The `:explorer` format is special, and will build an Explorer dataframe
+      in case the `:explorer` dependency is installed.
+
+  ## Examples
+
+  With a plain query:
+
+      iex> req = Req.new() |> ReqCh.attach()
+      iex> Req.post!(req, clickhouse: "SELECT number from system.numbers LIMIT 3").body
+      "0\\n1\\n2\\n"
```

Comment:
Yeah so [similar to ReqAthena discussion](https://github.com/livebook-dev/req_athena/pull/40#discussion_r1753663438), this really feels like better achieved with:

```elixir
iex> ReqCH.query!("SELECT ...")
iex> req = ReqCH.new(hostname: "clickhouse.cloud", retry: :transient)
iex> ReqCH.query!(req, "SELECT ...", retry: false)
```

Below is intentionally weird and unfair to this plugin because it is true for any plugin:

```elixir
req = Req.new() |> ReqCh.attach()
Req.get!(req, url: "https://httpbin.org", scheme: "http")
```

Just to be clear, this is totally on me FWIW.